utils/firestore.js

```diff
@@ -437,14 +437,16 @@ const removeParticipantsDataDestruction = async () => {
                 : 0
 
             if (participant[destroyDataCategoricalCId] === requestedAndSignCId || timeDiff > millisecondsInTwoDays) {
+                let hasRemovedField = false;
                 const fieldKeys = Object.keys(participant)
                 const participantRef = doc.ref;
                 fieldKeys.forEach(key => {
-                    if (!stubRecordList.includes(key)) {
+                    if (!stubFieldArray.includes(key)) {
                         batch.update(participantRef, { [key]: admin.firestore.FieldValue.delete() });
+                        hasRemovedField = true
```

Comment:
Maybe swap `count++` with `hasRemovedField = true` and get rid of `hasRemovedField`